executor/insert_common.go

```diff
@@ -747,6 +748,11 @@ func setDatumAutoIDAndCast(ctx sessionctx.Context, d *types.Datum, id int64, col
 	if err == nil && d.GetInt64() < id {
 		// Auto ID is out of range, the truncated ID is possible to duplicate with an existing ID.
 		// To prevent updating unrelated rows in the REPLACE statement, it is better to throw an error.
+		sc := ctx.GetSessionVars().StmtCtx
+		var insertPlan *core.Insert = sc.GetPlan().(*core.Insert)
```

Comment:
better to check if the type cast is ok to avoid panic `insertPlan, ok := sc.GetPlan().(*core.Insert)`.

Reply:
/cc @tangenta please help decide the behaviour when AUTO_ID reaches limit

Reply:
thanks, done.